src/vivarium/framework/components/manager.py

```diff
@@ -70,19 +89,54 @@ def setup_components(self, builder, configuration):
 
 
 class ComponentInterface:
+    """The component manager interface made available from the builder."""
 
     def __init__(self, component_manager: ComponentManager):
         self._component_manager = component_manager
 
     def add_components(self, components: Sequence):
+        """Register new components with the system. Components are setup after
+        managers.
+
+        Parameters
+        ----------
+        components:
+          Instantiated components to register
+        """
         self._component_manager.add_components(components)
 
     def get_components(self, component_type: str):
         return self._component_manager.get_components(component_type)
 
 
 def _setup_components(builder, component_list: Sequence, configuration: ConfigTree) -> Sequence:
-
+    """Helper method for configuring and setting up a list of components.
+
+    This function first loops over `component_list` and applies configuration
+    defaults if present.  The application entails updating the `configuration`
+    object. Then, the list is looped over until empty, with components being
+    popped and setup. Because of the side effect below, new components can be
+    added. This necessitates the while loop and also another check of
+    configuration.
+
+    Importantly, this function is called by the ComponentManager and it is
```

Comment:
use rst references where possible (e.g. :class:`ComponentManager`).  Anything defined in the module only needs a short name for the link to work properly.

Reply:
I nabbed everything in this module, but I am hesitant to go further. Maybe in the module level docstring. But references on classes to, say, lifecycle stages -- if we make it an rst reference it will render nice in sphinx but on the command line you'd see the whole rst syntax, right ?